lua/mason-core/spawn.lua

```diff
@@ -85,12 +78,14 @@ setmetatable(spawn, {
                 }, cmd)
             end
 
-            local _, exit_code, signal = a.wait(function(resolve)
-                local handle, stdio, pid = process.spawn(cmd, spawn_args, resolve)
+            local _, exit_code, signal = a.wait(vim.schedule_wrap(function(resolve)
```

Comment:
Preferably we shouldn't have to schedule to Neovim's main loop, these functions should be able to execute outside of it. This is pretty much exactly what I was thinking though, will just have to think about a way to only schedule when there's a need to invoke exepath (there's an `is_executable` call some lines above that I think we can repurpose somehow)

Reply:
I see. I'll give it some thought, but I'll hasten to report that `is_executable` actually returns true even for cmd that would cause `process.spawn` to fail.